doc/arch/adf-006-register-serialization-format.md

```diff
@@ -0,0 +1,101 @@
+        "C22"
+      ],
+      "dioceses": [
+        "CE23"
+      ],
+      "headteacher": "Mr T Wilson",
+      .....
+    }
+
+An Entry refers to an Item by the Item's SHA-256 hash and also contains a timestamp recording when the Item was added to the Register.
+
+The algorithm for computing the hash involves canonicalizing the JSON by removing all whitespace between elements i.e. not in a quoted string, calculating the SHA-256 hash on the JSON encoded as a UTF-8 string and expressed as a Hex string.
+
+The other key feature of a Register is its Root Hash which is the root element of the Merkle tree based on the sequence of Entries. The Root Hash can be used in combination with a Proof composed of a number of other hash values to verify a single Entry.
+
+Register data is currently provided by the Register Design Authority usually in TSV format e.g.
+
+    402378[tab]Abertillery Learning Community[tab][tab]677[tab][tab][tab]  [tab][tab][tab][tab]30[tab][tab][tab][tab][tab][tab]2016-09-01[tab]
+
+The TSV does not fully represent a Register in that the timestamps of the Entries and the Root Hash are not included.
+
+A Register is not the same thing as the Openregister-java application. It can be represented in an external, human readable format making the structure of the Register explicit. We call this the Register Serialization Format (RSF).
+
+The RSF can also be used to clone an existing Register as a whole or in part.
+
+Registers can be large - the address register currently has c34 million Entries and the food-premises-rating register c500,000. So the RSF should not need to be parsed as a whole before processing.
+
+### Validation
+
+To be valid:
+
+- All Items should be referred to by at least one Entry. There should be no Orphan Items.
+- All Entries should refer to an Item. In theory >1 Entry can point to the same Item. Where the RSF represents part of a Register, the Item may be part of the existing Register.
+
+
+## Decision ##
+
+- The requirement not to require parsing of the whole RSF rules out JSON.
+- XML can be streamed but there are security objections to using it.
+- The RSF will comprise Commands on separate lines, being one of those listed below.
+- Each Command comprises the Command name with arguments separated by tabs.
+- The Commands are applied to the register in the order they appear in the RSF
+- RSF must be encoded as UTF-8.
+
+### add-item
+
+#### arguments
+
+- a canonicalized JSON string
+
+Example
+
+    add-item[tab]{"address":"5T8EXT7N","denominations":["C22"],"dioceses":["CE23"],"headteacher":"Mr T Wilson","maximum-age":"11","minimum-age":"3","name":"Sir John Cass's Foundation Primary School","school":"100000","school-authority":"201","school-gender":"M","school-phase":"PRI","school-type":"2","website":"http://www.sirjohncassprimary.org"}
+
+#### Notes
+
+- add-item must appear before the corresponding append-entry
+
+### append-entry
+
+#### arguments
+
+- a timestamp in UTC in ISO 8601 format
```

Comment:
Can we use RFC 3339 like the [specification does](http://openregister.github.io/specification/#timestamp-datatype)? RFC 3339 is a much-restricted subset of ISO 8601 that fits our use case here. Compared to ISO 8601, RFC 3339:

 - only describes instants (not other things like a date without a corresponding time-of-day)
 - is described in a freely-available document.

Also, can we specify that the timestamp always uses the `Z` suffix rather than any other timezone?